src/App.js

```diff
@@ -5,27 +5,19 @@ import DopplerIntlProvider from './DopplerIntlProvider';
 import { FormattedMessage } from 'react-intl';
 import Header from './components/Header/Header';
 import Footer from './components/Footer/Footer';
-import axios from 'axios';
-import { HttpDopplerLegacyClient } from './services/doppler-legacy-client';
-import { OnlineSessionManager } from './services/session-manager';
 
 class App extends Component {
-  constructor(props) {
-    super(props);
+  constructor({ locale, dependencies: { sessionManager } }) {
+    super();
 
     this.updateSession = this.updateSession.bind(this);
 
-    this.sessionManager =
-      (props.dependencies && props.dependencies.sessionManager) ||
-      new OnlineSessionManager(
-        (props.dependencies && props.dependencies.dopplerLegacyClient) ||
-          new HttpDopplerLegacyClient(axios, process.env.REACT_APP_API_URL),
-        process.env.REACT_APP_DOPPLER_LEGACY_KEEP_ALIVE_MS,
-      );
+    /** @type { import('./services/session-manager').SessionManager } */
```

Comment:
What does this line?

Reply:
Oh, sorry, I thought that I have added a clarificatory comment, but it seems that something went wrong. 

As you have seen, _sessionManager_ instance is being injected using props, so IDE cannot determine the type of it and provide us a nice intellisense experience.

Since it is a JavaScript file and not a TypeScript one, we need using JSDoc to specify the type. See more information here: https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#import-types

Reply:
Does it make sense?